### PR TITLE
Bylined interfaces

### PR DESCRIPTION
**Pull request by legionth**
According to #7

### CODE REVIEW

File: src/UI/Component/Tree/Node/Factory.php [hunk 1]
```diff
@@ -41,4 +41,26 @@ interface Factory
...
 	public function simple(string $label, Icon $icon=null): Simple;
 
+    /**
+     * ---
+     * description:
+     *   purpose: >
+     *     The Bylined Node is an entry containing additional information about
+     *     the node.
+     *   composition: >
+     *     It consists of a string-label, a byline an optional Icon.
```
Comment (mjansenDatabay) [inline src/UI/Component/Tree/Node/Factory.php]: and an

File: src/UI/Component/Tree/Node/Factory.php [hunk 2]
```diff
@@ -41,4 +41,26 @@ interface Factory
...
+     *   effect: >
+     *     This node is a simple node with an additional string-byline.
+     * rules:
+     *   usage:
+     *      1: >
+     *        A Byline Node SHOULD when there is a need to display a byline
```
Comment (mjansenDatabay) [inline src/UI/Component/Tree/Node/Factory.php]: SHOULD be used?

File: src/UI/Component/Tree/Node/Bylined.php [hunk 3]
```diff
@@ -0,0 +1,20 @@
+<?php
+declare(strict_types=1);
+
+/* Copyright (c) 2019 Nils Haagen <nils.haagen@concepts-and-training.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Component\Tree\Node;
+
+/**
+ * This describes a tree node with an byline with additional information
```
Comment (mjansenDatabay) [inline src/UI/Component/Tree/Node/Bylined.php]: with a byline providing additional ...

File: src/UI/Component/Tree/Node/Bylined.php [hunk 4]
```diff
@@ -0,0 +1,20 @@
+<?php
+declare(strict_types=1);
+
+/* Copyright (c) 2019 Nils Haagen <nils.haagen@concepts-and-training.de> Extended GPL, see docs/LICENSE */
```
Comment (mjansenDatabay) [inline src/UI/Component/Tree/Node/Bylined.php]: Niels Theen

File: src/UI/Component/Tree/Node/Factory.php (hunk 2 above)
Comment (legionth) [reply src/UI/Component/Tree/Node/Factory.php]: That is exactly the question is it a MUST or a SHOULD do we want to enforce it?

Comment (mjansenDatabay) [reply src/UI/Component/Tree/Node/Factory.php]: I did not question the SHOULD ;-). The `be used` was missing :).

**Comment by mjansenDatabay** [comment]
LGTM!

Please link the respective feature wiki page (in the description, as reason for this new element) when opening the PR against the ILIAS `trunk`.

Could you also provide a screenshot of your current feature implementation (forum) in the PR's description?

**Comment by legionth** [comment]
https://github.com/ILIAS-eLearning/ILIAS/pull/2043

**Pull request closed without merging**